htlcswitch/link.go

```diff
@@ -1151,3 +1195,27 @@ func (l *channelLink) sendHTLCError(rHash [32]byte,
 		Reason: reason,
 	})
 }
+
+// waitForChannelReestablish waits for channel synchronization message to
+// be received from another side and handled.
+func (l *channelLink) waitForChannelReestablish() error {
+	// Before we launch any of the helper goroutines off the channel link
+	// struct, we'll first ensure proper adherence to the p2p protocol. The
+	// channel reestablish message MUST be sent before any other message.
+	select {
+	case msg := <-l.upstream:
+		if msg, ok := msg.(*lnwire.ChannelReestablish); ok {
+			l.handleUpstreamMsg(msg)
+		} else {
+			return errors.New("very first message between nodes " +
+				"for channel link should be reestablish message")
+		}
+	// In order to avoid blocking indefinitely, we'll give the other peer
+	// an upper timeout of 5 seconds to respond before we bail out early.
+	case <-time.After(time.Second * 5):
```

Comment:
Hmm, with travis in mind, I'm not sure 5 seconds will be sufficient. Let's wait to make a judgement until this is rebased so it can build on top of the latest master branch cleanly.